src/pages/index.js

```diff
@@ -40,89 +50,222 @@ const enableValidation = (config) => {
 }
+
+cardListData()
+  .then((data) => {
+    console.log("data", data);
+    const cardElement = data.map((cardData) => createCard(cardData));
+    cardList.renderItems(cardElement);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+/* ------------------------------ cardAddPopup ------------------------------ */
+
+const cardAddPopup = new PopupWithForm({
+  popupSelector: "#card-add-modal",
+  handleFormSubmit: handleCardAddSubmit,
+});
+cardAddPopup.setEventListeners();
+
+cardAddButton.addEventListener("click", () => {
+  cardAddPopup.open();
+});
+
 function handleCardAddSubmit(formValues) {
   const name = formValues.title;
   const link = formValues.link;
-  renderCard({ name, link });
+  cardAddPopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+
+    .then((card) => {
+      const cardElement = createCard(card);
+      cardList.addItem(cardElement);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardAddPopup.renderLoading(false);
+    });
+
   cardAddPopup.close();
   cardAddForm.reset();
   formValidators[cardAddForm.getAttribute("id")].disableButton();
 }
-/* -------------------------------------------------------------------------- */
-/*                               Event Listeners                              */
-/* -------------------------------------------------------------------------- */
-/* ------------------------------ Profile Edit ------------------------------ */
-profileEditButton.addEventListener("click", () => {
-  fillProfileForm();
-  //editFormValidator.resetValidation();
-  formValidators[profileEditForm.getAttribute("id")].resetValidation();
-  profileEditPopup.open();
-});
-/* ---------------------------------- Card ---------------------------------- */
-cardAddButton.addEventListener("click", () => {
-  //cardAddFormValidator.resetValidation();
-  cardAddPopup.open();
+/* ----------------------------- cardImagePopup----------------------------- */
+const cardImagePopup = new PopupWithImage("#image-modal");
+cardImagePopup.setEventListeners();
+
+function handleImageClick(name, link) {
+  cardImagePopup.open(name, link);
+}
+
+/* --------------------------- DeleteConfirmPopup --------------------------- */
+const deleteConfirmPopup = new PopupWithConfirmation({
+  popupSelector: "#confirmation-modal",
+  handleFormSubmit: handleDeleteClick,
 });
+deleteConfirmPopup.setEventListeners();
+
+function handleDeleteClick(card) {
+  deleteConfirmPopup.open();
+  deleteConfirmPopup.confirmDelete(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.removeCard();
+        deleteConfirmPopup.close();
+      })
+      .catch((err) => {
+        console.error("Card deletion API failed:", err);
+      });
+  });
+}
+
+/* -------------------------------- LikeCard -------------------------------- */
+function handleLikeClick(card) {
+  const currentStatus =
+    card._likeButton.classList.contains("card__like_active");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle

Reply:
Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.